### ISSUE TITLE
[DatePicker] JAWS announce the wrong word when using Right Arrow key to listen the spelling of the word in calendar

### ISSUE THREAD

**Issue by jaimiegorman** [carbon-design-system/carbon#7668]
===== ENVIRONMENT =====
IBM CI 162 Checkpoint Violation:  4.1.2 Name,Role,Value

IBM Guidance: https://www.ibm.com/able/guidelines/ci162/keyboard.html

https://www.ibm.com/able/guidelines/ci162/name_role_value.html

Test Tool Type: Manual test + Screen Reader JAWS (AVT3)
JAWS version: 2021.2011.16 ILM
Browser: Firefox 83.0 (64-bit)

===== DEFECT DESCRIPTION =====
JAWS announce wrong words when using Right Arrow key to listen the spelling of the word in calendar.

It will be read letters "SUN" associated with the word "November".
If press three Right Arrow key again, it will read letters "MON" associated with the word "November". And so on,
read letters "TUE" associated with the word "Igol?",
read letters "WED" associated with the word "daughter?",
read letters "THU" associated with the word "uniform?",
read letters "FRI" associated with the word "Indian?",
read letters "SAT" associated with the word "Tigol?".

These extra words are wrong, they will interfere with the understanding of the blind.

=====STEPS TO REPRODUCE =====
1. Start Screen Reader JAWS
2. Login to RTWA employee home page: https://test.wh-rtw.dev.watson-health.ibm.com/employee/
3. Access RTWA employee homepage, click Check-in bitmap
4. Use keyboards(Tab key and Down Arrow key) to navigate three calendar widgets
5. Click Enter key to open the calendar
6. Press Down Arrow key to select the current day.
7. Press Up Arrow key to select Week name "Sun Mon Tue Wed Thu Fri Sat",see image
7. Press three Right Arrow key to listen the spelling of the word.

===== EXPECTED RESULT =====
Don't announce extra wrong words or provide correct word for weekday when using Right Arrow key to listen the spelling of the word in calendar.

![image](https://user-images.githubusercontent.com/78103524/106034547-08e29400-60a1-11eb-96c0-2d7d2e61d022.png)

**Comment by dakahn** [comment]
I'm unable to reproduce this bug in Firefox v103.0.2 on Windows 11 using JAWS v2022.2207.25ILM. 

Entering reading mode inside the box has the expected behavior. Since the user is on a form input for date the expectation isn't that they would read through the dialog below like a modal or popover. The "calendar" with headings that represent the days of the week is a visual affordance. I'm not even sure how you'd access them with JAWS virtual cursor. 

Im closing as fixed/false report. But I'm happy to open again if this problem still persists in user testing.
